torch/csrc/jit/source_range.h

```diff
@@ -7,29 +7,108 @@
 namespace torch {
 namespace jit {
 
+// This structure contains information regarding the original source file
+// associated with a SourceRange. In particular, it records the original file
+// name, as well as the offset of source_ with respect to the original source
+// file.
+//
+// That is, if a SourceRange describes a location within a function, source_ 
+// will contain only the source code for that function, and not the whole file
+// due to the semantics of Python's inspect module. To recover original source
+// file line information, we store away file_ofset_, which is the line offset
+// of source_ with respect to the beginning of its source file.
+struct File {
+  File(
+      std::shared_ptr<std::string> source,
+      std::string filename,
+      size_t file_offset)
+      : source_(std::move(source)),
+        filename_(std::move(filename)),
+        file_offset_(file_offset) {
+    size_t pos = 0;
+    do {
+      line_len_prefix_sum_.push_back(pos);
```

Comment:
line_starting_offset_? line_len_prefix_sum_ is a description of how to compute this, but obscures what it is.